webapp/portlet/src/main/webapp/vue-apps/space-settings/components/SpaceSettings.vue

```diff
@@ -24,19 +25,21 @@ export default {
   }),
   created() {
     // add external components
-    const externalComponents = extensionRegistry.loadComponents('external-space').map(component => component.componentOptions.componentImpl);
+    const externalComponents = extensionRegistry.loadComponents('external-apps-space-settings').map(component => component.componentOptions.componentImpl);
     this.spaceExternalSettings.push(...externalComponents);
-
-    document.addEventListener('addSpaceSettingsExternalComponents', (event) => {
-      if (event && event.detail) {
-        this.spaceExternalSettings.push(event.detail.componentImpl);
-      }
-    });
-
     document.addEventListener('hideSettingsApps', () => this.displayed = true);
     document.addEventListener('showSettingsApps', () => this.displayed = true);
   },
   mounted() {
+    document.addEventListener('component-external-apps-space-settings-space-settings-updated', (event) => {
+      if (event && event.detail ) {
+        this.$spaceService.getSpaceApplications(this.spaceId).then(applications => {
+          if (applications.find(app => app.id === event.detail.componentOptions.appId) || !event.detail.componentOptions.appId) {
+            this.spaceExternalSettings.push(event.detail.componentOptions.componentImpl);
+          }
+        });
+      }
+    });
     this.$nextTick().then(() => this.$root.$applicationLoaded());
```

Comment:
> We should retrieve Space Applications only once IMO outside this event. This call will be used to retrieve the same space's applications for each **space settings extension**, isn't it ?

```suggestion
    this.$spaceService.getSpaceApplications(this.spaceId)
      .then(applications => {
        document.addEventListener('component-external-apps-space-settings-space-settings-updated', (event) => {
          if (event && event.detail ) {
            if (applications.find(app => app.id === event.detail.componentOptions.appId) || !event.detail.componentOptions.appId) {
              this.spaceExternalSettings.push(event.detail.componentOptions.componentImpl);
            }
          });
        }
        // add external components
        const externalComponents = extensionRegistry.loadComponents('external-space').map(component => component.componentOptions.componentImpl);
        this.spaceExternalSettings.push(...externalComponents);
      })
      .finally(() => this.$root.$applicationLoaded());
```

Reply:
To place in `created()` instead of `mounted()` to retrieve applications early in Page rendering phases.